sql/analyzer/resolve_tables.go

```diff
@@ -346,6 +346,7 @@ func transferProjections(ctx *sql.Context, from, to *plan.ResolvedTable) *plan.R
 	}
 
 	if _, ok := toTable.(sql.FilteredTable); ok {
+		toTable.(sql.FilteredTable).HandledFilters(filters)
```

Comment:
Something is off here. `HandledFilters` seems like a non-mutating method when I look at the analyzer? And `WithFilters` is mutating? So it doesn't seem like Dolt tables should only mutate for the `HandledFilters` call? I'd think Dolt commit table should answer `HandledTables` and mutate on `WithFilters` I guess, so not really a GMS problem unless the interfaces are wrong.

Reply:
Max is right. This is `HandledFilters`, it's supposed to return the subset of predicate expressions that can be pushed down to the table. The use here is a no-op.

If the memory table implementation is doing something different, that's a bug.